syntax = "proto3";

package indentcorp.platform.notid.v1alpha1;

import "google/protobuf/struct.proto";
import "google/protobuf/timestamp.proto";


service Message {
  rpc create (MessageRequest) returns (MessageResponse);
}

message Platform {
    enum Name {
        KAKAO = 0;
    }
    Name name = 1;
    google.protobuf.Struct context = 2;
}

message TargetSchedule {
  google.protobuf.Timestamp send_time = 1;
}


message RangeSchedule {
  google.protobuf.Timestamp send_start_time = 1;
  google.protobuf.Timestamp send_end_time = 2;
}


message MessageRequest {
    Platform platform = 1;
    google.protobuf.Struct metadata = 2;
    oneof schedule {
      TargetSchedule target_schedule = 3;
      RangeSchedule range_schedule = 4;
    }
    oneof payload {
      string string_payload = 5;
      google.protobuf.Struct dict_payload = 6;
    }
}

message MessageResponse {
  string message_id = 1;
}
